### DISCUSSION TITLE
how to clone a repository to github

### DISCUSSION THREAD

**Discussion by RUBIYASULTHANA** [community/community#125632] [New to GitHub]
how to clone a repository to github

**Comment by MrShadowRIFAT** [comment]
Hi there,

To clone a repository from GitHub to your local machine, follow these steps:

1. **Install Git**: Ensure that Git is installed on your machine. You can download it from [git-scm.com](https://git-scm.com/).

2. **Open Terminal or Command Prompt**: Open your terminal (on macOS or Linux) or Command Prompt (on Windows).

3. **Navigate to the Desired Directory**: Use the `cd` command to navigate to the directory where you want to clone the repository.
    ```sh
    cd path/to/your/directory
    ```

4. **Get the Repository URL**: Go to the GitHub repository page you want to clone. Click the green "Code" button and copy the URL (either HTTPS or SSH).

5. **Clone the Repository**: Use the `git clone` command followed by the repository URL. For example:
    ```sh
    git clone https://github.com/username/repository.git
    ```
    or if using SSH:
    ```sh
    git clone git@github.com:username/repository.git
    ```

Here's a more detailed step-by-step example:

### Step-by-Step Example

1. **Install Git**:
    - Download and install Git from [git-scm.com](https://git-scm.com/).
    - Follow the installation instructions for your operating system.

2. **Open Terminal or Command Prompt**:
    - On macOS: Open Terminal.
    - On Windows: Open Command Prompt or Git Bash.
    - On Linux: Open your preferred terminal emulator.

3. **Navigate to Your Directory**:
    ```sh
    cd ~/projects
    ```

4. **Get the Repository URL**:
    - Go to the repository page on GitHub (e.g., `https://github.com/username/repository`).
    - Click the green "Code" button.
    - Copy the URL from the dropdown (HTTPS or SSH).

5. **Clone the Repository**:
    ```sh
    git clone https://github.com/username/repository.git
    ```

6. **Verify the Clone**:
    - After cloning, navigate into the repository folder:
      ```sh
      cd repository
      ```
    - List the files to verify the contents:
      ```sh
      ls
      ```

### Using SSH (Optional)
If you prefer using SSH for authentication, you need to set up SSH keys:

1. **Generate an SSH Key**:
    ```sh
    ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
    ```
    - Follow the prompts and save the key to the default location.

2. **Add the SSH Key to Your GitHub Account**:
    - Copy the SSH key to your clipboard:
      ```sh
      cat ~/.ssh/id_rsa.pub
      ```
    - Go to [GitHub SSH and GPG keys settings](https://github.com/settings/keys).
    - Click "New SSH key" and paste the key.

3. **Clone Using SSH**:
    ```sh
    git clone git@github.com:username/repository.git
    ```

Let me know if you need further assistance or if you have any other questions!

Best regards,  
[MrShadowRIFAT](https://rifat.website)

**Comment by resort3779** [comment]
This is a good [video](https://www.youtube.com/watch?v=8Dd7KRpKeaE) to start

**Comment by AsmitaMishra24** [comment]
Hey @RUBIYASULTHANA!

I've provided two methods to choose from:

If you want to contribute to the project and make PRs, then follow the optional steps also

Method 1: GitHub Desktop:-

To clone a repository from GitHub using the GitHub Desktop method, follow these steps:

Step 1: Install GitHub Desktop
- Go to the GitHub Desktop download [page.](https://desktop.github.com/)
- Download the appropriate version for your operating system (Windows or macOS).
- Open the downloaded file and follow the installation instructions.

Step 2: Fork the Repository(Optional)
- Open your web browser and navigate to the GitHub repository you want to clone.
- Click the Fork button at the top right corner of the repository page. This creates a copy of the repository in your own GitHub account.
![fork](https://github.com/community/community/assets/146121869/7668bbaf-b1da-4414-94fa-603af192d298)

Step 3: Clone the Forked Repository
- After forking the repository, you will be redirected to your forked repository's page.
- Click the Code button located above the list of files in the repository.
![code](https://github.com/community/community/assets/146121869/a92e5326-fa26-43e9-bd0c-3b8f1bb1a144)

- In the dropdown menu, click on Open with GitHub Desktop.
![clone](https://github.com/community/community/assets/146121869/5051ce08-0dd2-4514-b346-09e779d8cfbe)

- If prompted, allow the web browser to open GitHub Desktop.
![gitdesktop](https://github.com/community/community/assets/146121869/a4e420d0-2d65-4865-8641-3070cc4d6448)

- GitHub Desktop will open and show a dialog to clone the repository. Choose the local path where you want to save the repository on your computer.
- Click the Clone button to start cloning the repository to your local machine.

OR

Method 2: ZIP File:-

To clone a repository from GitHub using the ZIP download method, follow these steps:

Step 1: Fork the Repository (Optional):
- If you want to contribute to the repository, you can fork it by clicking the "Fork" button on the top right corner of the repository page. This will create a copy of the repository in your own GitHub account.
![fork](https://github.com/community/community/assets/146121869/f912ecc7-a0ac-4d19-b6d7-63b9d45e2ab9)

Step 2: Download ZIP File:
- Open your web browser and navigate to the GitHub repository you want to clone.
- Click on the repository name to enter the repository.
- Click the "Code" button located above the list of files in the repository.
![code](https://github.com/community/community/assets/146121869/4009e4a8-ec00-4be8-918a-3d58a6254764)

- In the dropdown menu, click on "Download ZIP". This will download the repository as a ZIP file to your local machine.
![clone](https://github.com/community/community/assets/146121869/8ad277f4-a939-4446-9b88-ddd5756ca45b)

Step 3: Extract ZIP File:
- Once the ZIP file is downloaded, locate it in your file explorer.
- Right-click on the ZIP file and select "Extract All..." or a similar option based on your operating system.
![extract](https://github.com/community/community/assets/146121869/3c593f89-33d3-4517-a11b-249ff88af1dd)

- Choose a destination folder for the extracted files and click "Extract".

Step 4: Run the Code:
- Open the extracted folder.
- Look for any setup instructions or README files for guidance on how to run the code.
- Follow the instructions provided to set up your environment and run the code accordingly.

I hope this will be helpful.
